docs/src/app/(public)/(content)/react/utils/use-render/demos/render-callback/css-modules/index.tsx

```diff
@@ -14,37 +15,28 @@ interface CounterProps extends useRender.ElementProps<CounterState> {
 }
 
 function Counter(props: CounterProps) {
-  const {
-    render = <button />,
-    className,
-    onClick,
-    'aria-label': ariaLabel,
-    ...otherProps
-  } = props;
+  const { render = <button />, ...otherProps } = props;
   const [count, setCount] = React.useState(0);
   const odd = count % 2 === 1;
   const state = React.useMemo(() => ({ odd }), [odd]);
 
-  const handleClick = (event: React.MouseEvent) => {
-    setCount((prev) => prev + 1);
-    onClick?.(event);
-  };
-
   const { renderElement } = useRender({
     render,
     state,
-    props: {
-      className: `${styles.Button} ${className ?? ''}`,
-      type: 'button',
-      children: (
-        <React.Fragment>
-          Counter: <span>{count}</span>
-        </React.Fragment>
-      ),
-      onClick: handleClick,
-      'aria-label': ariaLabel ?? `Count is ${count}, click to increase.`,
-      ...otherProps,
-    },
+    props: mergeProps(
+      {
+        className: styles.Button,
+        type: 'button',
+        children: (
+          <React.Fragment>
+            Counter: <span>{count}</span>
+          </React.Fragment>
+        ),
+        onClick: () => setCount((prev) => prev + 1),
+        'aria-label': `Count is ${count}, click to increase.`,
+      },
+      otherProps,
+    ),
```

Comment:
`otherProps` goes first

Reply:
It's neater with Prettier to place the variable first

Reply:
> Devs nowadays may be more familiar with object spread pattern that works from left to right than Object.assign that mergeProps currently mimics.

This was my initial thought as well, if I spread this would be the order.

Reply:
Also for props that should replace (like aria-label in the demo), you want other props to override, so it needs to comes second, no?

Reply:
In the `mergeReactProps` implementation, the ones on the left take precedence. So `aria-label` from the user will override our `aria-label` when it's specified in the object later

Re-thinking it, I guess it does make sense for ones specified later in the arguments to override earlier ones instead. Prettier formatting is only a bit worse

Reply:
I feel strongly that we should change this then, as Michal mentioned, object spread, Object.assign etc all work the other way around. Are we ok with changing this? If yes, I can apply the change as part of this PR.

Reply:
I opened a new PR for this, there were too many changes: https://github.com/mui/base-ui/pull/1533